src/Storefront/Test/Page/Checkout/FinishPageTest.php

```diff
@@ -39,6 +41,15 @@ public function testFinishPageLoading(): void
         $context = $this->createSalesChannelContextWithLoggedInCustomerAndWithNavigation();
         $orderId = $this->placeRandomOrder($context);
         $request = new Request([], [], ['orderId' => $orderId]);
+        $dispatcher = $this->getContainer()->get('event_dispatcher');
+        $eventWasThrown = false;
+        $criteria = new Criteria([$orderId]);
+
+        $listener = static function (CheckoutFinishPageOrderCriteriaEvent $event) use ($criteria, &$eventWasThrown): void {
+            static::assertSame($criteria->getIds(), $event->getCriteria()->getIds());
+            $eventWasThrown = true;
+        };
+        $dispatcher->addListener(CheckoutFinishPageOrderCriteriaEvent::class, $listener);
```

Comment:
there is a method in the test case this->addEventDispatcher. Please use that

Reply:
@shyim You mean $this->addEventListener();? In this class I don't have method addEventDispatcher

Reply:
Should be inside the IntegrationTestBehaviour trait

Reply:
@shyim I checked the whole repository and I didn't find any addEventDispatcher method. Are you sure it is correct name?

Reply:
sorry addEventListener 🙈

Reply:
@shyim In this way? https://github.com/shopware/platform/pull/2324/commits/d3b4100ee37f4e3e1dc764418d1c910f8b410e15